library/rsa.c

```diff
@@ -1550,10 +1550,17 @@ int mbedtls_rsa_rsassa_pss_sign( mbedtls_rsa_context *ctx,
         return( MBEDTLS_ERR_RSA_BAD_INPUT_DATA );
 
     hlen = mbedtls_md_get_size( md_info );
-    slen = hlen;
 
-    if( olen < hlen + slen + 2 )
+    /* Calculate the largest possible salt length. The salt length chosen will
+     * be the minimum of either the hash length or the key size minus the hash
+     * length minus 2 bytes. The salt length is always at least the hash length
+     * minus 2 bytes. This complies with FIPS 186-4 §5.5 (e) and RFC 8017
+     * (PKCS#1 v2.2) §9.1.1 step 3. */
```

Comment:
Please move the sentence “This complies …” before “The salt length is always at least …”, since the “at least” constraint isn't required by either FIPS or PKCS#1.

Reply:
Moved. Didn't want to imply that FIPS or PKCS#1 required the chosen minimum bound.